### PR TITLE
Module 4

### PR DESCRIPTION
**Pull request by shivaHanchate**
Hi Alex/Hugh/Jack,
Please review Module 4 project exercise.
Many Thanks
Shiva

### CODE REVIEW

File: .vagrant/rgloader/loader.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# This file loads the proper rgloader/loader.rb file that comes packaged
```
Comment (hugh-emerson) [inline .vagrant/rgloader/loader.rb]: If files are already being tracked by git, adding them to .gitignore is not enough. You can run e.g. `git rm --cached .vagrant/rgloader/loader.rb` to "delete" the file from the repository while keeping it locally.

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    # Install new python version and set as default    
+    pyenv install 3.8.3
+    pyenv global 3.8.3
+    python --version
+
+    # Download and Install poetry
+    curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+    #source $HOME/.poetry/env
```
Comment (hugh-emerson) [inline Vagrantfile]: Delete this commented out line.

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  SHELL
+
+  config.trigger.after :up do |trigger|
+    trigger.name = "Launching App"
+    trigger.info = "Running the TODO app setup script"
+    trigger.run_remote = {privileged: false, inline: "
+      # Install dependencies and launch
+      # <your script here>
```
Comment (hugh-emerson) [inline Vagrantfile]: Delete placeholder comment.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+description = "A simple ToDo app for your DevOps apprenticeship"
+authors = []
+
+[tool.poetry.dependencies]
+python = "^3.7"
+flask = "^1.1.2"
+python-dotenv = "^0.14.0"
+requests = "^2.25.1"
+pendulum = "^2.1.2"
+selenium = "^3.141.0"
```
Comment (hugh-emerson) [inline pyproject.toml:13]: selenium should also be in the dev-dependencies section

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      cd /vagrant
+      poetry install
+      poetry run flask run --host=0.0.0.0
```
Comment (hugh-emerson) [inline Vagrantfile]: This currently makes the "vagrant up" command hang while the server runs. You could instead start the Flask app as a background process on the VM by running: 
`nohup poetry run flask run --host=0.0.0.0 > app.log 2>&1 &`

Main benefit is Vagrant won't get confused about the state of the VM.

**Review by hugh-emerson — APPROVED** [review]
Looks good! I've left some minor comments but merge whenever you're ready.

File: .vagrant/rgloader/loader.rb (hunk 1 above)
Comment (shivaHanchate) [reply .vagrant/rgloader/loader.rb]: Deleted as advised

File: Vagrantfile (hunk 2 above)
Comment (shivaHanchate) [reply Vagrantfile]: Comment deleted

File: Vagrantfile (hunk 3 above)
Comment (shivaHanchate) [reply Vagrantfile]: placeholder deleted

File: pyproject.toml (hunk 4 above)
Comment (shivaHanchate) [reply pyproject.toml:13]: Added selenium = "^3.141.0" to dev dependencies

File: Vagrantfile (hunk 5 above)
Comment (shivaHanchate) [reply Vagrantfile]: amended as advised.

**Pull request merged**